go.mod

```diff
@@ -29,16 +29,16 @@ require (
 	go.opentelemetry.io/otel/exporters/metric/prometheus v0.13.0
 	go.opentelemetry.io/otel/exporters/trace/jaeger v0.13.0
 	go.opentelemetry.io/otel/sdk v0.13.0
-	golang.org/x/crypto v0.0.0-20201117144127-c1f2f97bffc9
-	k8s.io/api v0.21.0-beta.0
-	k8s.io/apimachinery v0.21.0-beta.0
-	k8s.io/client-go v0.21.0-beta.0
-	k8s.io/component-base v0.21.0-beta.0
-	k8s.io/klog/v2 v2.5.0
-	k8s.io/kubectl v0.21.0-beta.0
-	k8s.io/utils v0.0.0-20210111153108-fddb29f9d009
-	sigs.k8s.io/cluster-api v0.3.11-0.20210309152500-ef7a51b6d725
-	sigs.k8s.io/controller-runtime v0.8.3
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	k8s.io/api v0.21.0-beta.1
+	k8s.io/apimachinery v0.21.0-beta.1
+	k8s.io/client-go v0.21.0-beta.1
+	k8s.io/component-base v0.21.0-beta.1
+	k8s.io/klog/v2 v2.8.0
+	k8s.io/kubectl v0.21.0-beta.1
+	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10
+	sigs.k8s.io/cluster-api v0.0.0-20210325161731-7124a659accd
```

Comment:
this doesn't look right, should be v0.3.11-0......

Reply:
Since this is from the main branch, and we don't have a tagged release yet, we'll have to use `0.0.0` to specify a [pseudo version](https://golang.org/ref/mod#pseudo-versions). That being said, we should create a followup story to update to a release version once `0.4` is out. I've filed an [issue](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1274) to document the steps for upgrading dependencies. wdyt?

Reply:
This is also correct according to this link here https://golang.org/ref/mod#pseudo-versions